File: js/fib.js
let iter = 0, x = 0, y = 0;
let test = 0;

for (let i = 0; i < 1000000000; i++) {
	let x = 0;
	let y = 1;
	while (true) {
		iter++;
		if (iter % 2 == 0) {
			test += x;
		}
		else {
			test -= x;
		}
		let i = x + y;
		x = y;
		y = i;
		if (x > 999999999) {
			break;
		}
	}
}
console.log(`iter: ${iter} ${test}`)

File: nim/fib.nim
proc main() =
    var iter = 0
    var test = 0
    for i in 0 ..< 1000000000:
        var 
            prev = 0
            current = 1
        while prev <= 999999999:
            iter += 1 
            if iter mod 2 == 0:
                test += prev
            else:
                test -= prev
            current += prev
            prev = current - prev
    echo("Done in: ", iter, " Test: ", test)

main() 
